ui/src/app/settings/containers/ProxyForm/ProxyForm.js

```diff
@@ -0,0 +1,128 @@
+import { Formik, Form } from "formik";
+import React from "react";
+import { useSelector } from "react-redux";
+import * as Yup from "yup";
+
+import selectors from "app/settings/selectors";
+import Button from "app/base/components/Button";
+import Input from "app/base/components/Input";
+
+const ProxySchema = Yup.object().shape({
+  proxyType: Yup.string().required(),
+  httpProxy: Yup.string().when("proxyType", {
+    is: val => val === "externalProxy" || val === "peerProxy",
+    then: Yup.string()
+      .url("Must be a valid URL.")
+      .required("Please enter the proxy URL.")
+  })
+});
+
+const ProxyForm = () => {
+  const httpProxy = useSelector(selectors.config.httpProxy);
+  const proxyType = useSelector(selectors.config.proxyType);
+
+  return (
+    <Formik
+      initialValues={{
+        httpProxy,
+        proxyType
+      }}
+      onSubmit={(values, { setSubmitting }) => {
+        setTimeout(() => {
+          alert(JSON.stringify(values, null, 2));
+          setSubmitting(false);
+        }, 400);
+      }}
+      validationSchema={ProxySchema}
+    >
+      {({
+        handleBlur,
+        handleChange,
+        handleSubmit,
+        isSubmitting,
+        touched,
+        errors,
+        values
+      }) => (
+        <Form onSubmit={handleSubmit}>
+          <Input
```

Comment:
I created a new component for handling Formik fields to reduce most of the repetition below. See: https://github.com/canonical-web-and-design/maas-ui/blob/master/ui/src/app/settings/components/UserFormFields/UserFormFields.js#L22.